subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildTreeScopeServices.java

```diff
@@ -200,6 +209,25 @@ ModuleRepositoryCacheProvider createModuleRepositoryCacheProvider(
         return new ModuleRepositoryCacheProvider(persistentCaches, inMemoryOnlyCaches);
     }
 
+    ImmutableTransformWorkspaceServices createImmutableTransformWorkspaceServices(
```

Comment:
🤔 Note that this causes that the transformed artifacts between the same projects won't be re-used depending on the composite build project setup, since the root build directory - and therefore the cache location - might change.

Reply:
What do you think would be better option here? Should we move cache to build scope?

Reply:
I think it is okay. Build Scope would be another option, though you won't have re-use between participants of the composite build.